google-cloud-spanner/lib/google/cloud/spanner/session.rb

```diff
@@ -1265,6 +1270,15 @@ def session
         def ensure_service!
           raise "Must have active connection to service" unless service
         end
+
+        def merge_if_present hash, hash_to_merge
+          if hash.nil?
+            hash = hash_to_merge
+          else
+            hash = hash_to_merge.merge hash unless hash_to_merge.nil?
+          end
+          hash
```

Comment:
This line shouldn't be necessary, I think. The last expression(s) within the condition is returned by default.

Reply:
Had done it as statement within else is also conditional. But now have update the statement to handle the nil case too.